src/Plugin/resource/Field/PublicFieldInfo/PublicFieldInfoBase.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\restful\Plugin\resource\Field\PublicFieldInfo\PublicFieldInfoBase.
+ */
+
+namespace Drupal\restful\Plugin\resource\Field\PublicFieldInfo;
+
+use Drupal\restful\Exception\ServerConfigurationException;
+
+class PublicFieldInfoBase implements PublicFieldInfoInterface {
+
+  /**
+   * Default sections for the field information.
+   *
+   * @var array[]
+   */
+  protected static $defaultSections = array(
+    'info' => array(
+      'label' => '',
+      'description' => '',
+    ),
+    // Describe the data.
+    'data' => array(
+      'type' => NULL,
+      'read_only' => FALSE,
+      'cardinality' => 1,
+      'required' => FALSE,
+    ),
+    // Information about the form element.
+    'form_element' => array(
+      'type' => NULL,
+      'default_value' => '',
+      'placeholder' => '',
+      'size' => NULL,
+      'allowed_values' => NULL,
+    ),
+  );
+
+  /**
+   * Sections for the field information.
+   *
+   * @var array[]
+   */
+  protected $sections = array();
```

Comment:
so `category`sounds better I think. section for me has a visual connotation, rather then data.